crates/autopilot/src/infra/cli.rs

```diff
@@ -0,0 +1,8 @@
+use reqwest::Url;
+
+#[derive(Debug, clap::Parser)]
+pub struct Args {
+    /// The node RPC API endpoint.
+    #[clap(long, env)]
+    pub ethrpc: Url,
```

Comment:
Shouldn't this be deleted somewhere else since we don't want to configure it in multiple places do we?

Reply:
Since migrating all configuration parameters in one go is cumbersome, my idea was:
1. Add parameters to this file as we refactor things
2. Populate values of these parameters from the `arguments.rs` values somewhere in `run` function, and then use them going forwards.
3. Once all migrated, kill `arguments.rs` and use this file directly.

Example: https://github.com/cowprotocol/services/pull/2206/files#diff-3c53ab7c0731350f70c25e12902d084c2d65e99ff25f0db80c301722ecf7222cR633-R639

Although I am open to other suggestions how to migrate if there is easier option.

Reply:
I don't think we should add another `clap` configurable argument if it's literally a 1:1 replacement for an existing one that continues to exist (node_url). `ethrpc` is not even parsed from the command line (making it extra confusing for future maintainers of the code)

If you want to start the migration in this PR, I think `infra::cli::Args` should become a flattened part of `arguments::Arguments` (and node_url moved away from `shared::arguments::Arguments`), but I think it would be more consistent to create a separate PR that moves all of `arguments::Arguments` into `infra::cli::Args`. Otherwise we end up with two way of specifying arguments for this binary which again is confusing for future maintainers.